### PR TITLE
Add AVS operators table

### PR DESCRIPTION
**Pull request by vincenthongzy**
backend https://github.com/NethermindEth/restaking-dashboard-backend/pull/93

### CODE REVIEW

**Review by rubo — CHANGES_REQUESTED** [review]
The Share column title and items are misaligned. I know the design also looks like that, but let's fix it and left align.

image

File: src/@services/apiCall.js [hunk 1]
```diff
@@ -26,7 +26,8 @@ export default async function apiCall(endpoint, options) {
       'Content-Type': `application/json; charset=utf-8`,
       ...options?.headers
     },
-    method: options?.method?.toUpperCase() || 'GET'
+    method: options?.method?.toUpperCase() || 'GET',
+    signal: options?.signal
```
[the remark this answers is not in the document]
Comment (vincenthongzy) [reply src/@services/apiCall.js]: rebased accordingly

File: src/@services/AVSService.js [hunk 2]
```diff
@@ -28,20 +28,22 @@ export default class AVSService {
...
+        search,
+        sort
+      },
+      signal
     });
 
     if (response.ok) {
       return await response.json();
+    } else {
+      throw new Error(await response.json());
```
[the remark this answers is not in the document]
Comment (vincenthongzy) [reply src/@services/AVSService.js:48]: will update after you merge #129

File: src/avs/AVSDetailsOperatorsTab.jsx [hunk 3]
```diff
@@ -66,49 +71,310 @@ export default function AVSDetailsOperatorsTab({ operators }) {
...
+        <Input
+          className="w-full lg:w-[40%]"
+          classNames={{
+            inputWrapper:
+              'border border-outline data-[hover=true]:border-foreground-1',
+            input: 'placeholder:text-foreground-2'
+          }}
+          endContent={<span className="material-symbols-outlined">search</span>}
+          onChange={handleInputChange}
+          placeholder="Search by address/name with 3 characters or more..."
```
[the remark this answers is not in the document]
Comment (vincenthongzy) [reply src/avs/AVSDetailsOperatorsTab.jsx:226]: updated to have same phrasing as AVS List

**Comment by vincenthongzy** [comment]
> The Share column title and items are misaligned. I know the design also looks like that, but let's fix it and left align.
> 
> image

centered instead

**Comment by rubo** [comment]
I have mixed feelings about this center alignment, but okay, let's leave it as is.

image

**Comment by vincenthongzy** [comment]
> I have mixed feelings about this center alignment, but okay, let's leave it as is.
> 
> image

I made it the same as AVS List, left-aligned

**Review by Zishan-7 — APPROVED** [review]
LGTM

File: src/avs/AVSDetailsOperatorsTab.jsx [hunk 4]
```diff
@@ -66,49 +72,293 @@ export default function AVSDetailsOperatorsTab({ operators }) {
...
+          radius="sm"
+          type="text"
+          value={state.search ?? ''}
+          variant="bordered"
+        />
+      </div>
+
+      <Table
+        aria-label="List of operators registered for AVS"
+        className="overflow-x-scroll"
```
Comment (rubo) [inline src/avs/AVSDetailsOperatorsTab.jsx:237]: This is redundant and shows a scrollbar always. Removing.

```suggestion
```

**Review by rubo — APPROVED** [review]

**Pull request merged**